databuilder/__main__.py

```diff
@@ -43,6 +45,7 @@ def main(args, environ=None):
 
     if options.which == "generate-dataset":
         if options.dsn:
+            assert options.backend != EXPECTATIONS_BACKEND_PLACEHOLDER
```

Comment:
This is fine in this context, but as a general rule I think you're better off using identity checks with sentinel objects (`options.backend is not EXPECTATIONS_BACKEND_PLACEHOLDER`) because of the possibility of the LHS doing funny-business with equality overloading. Plus you can't meaningfully talk about the value of sentinel objects anyway: their identity _is_ their entire role.

Reply:
Oh, thank you -- you're quite right. That has not occurred to me. I think the risk here is v low, though, so I'm going to leave it as it's merged. I wouldn't be at all offended if you changed it in passing later (or I might do the same).

Reply:
I think the risk is exactly zero here! I certainly don't think it needs changing. I just thought you might be interested in the more general stylistic point.